src/Graphics/Effect/Effect.cs

```diff
@@ -940,41 +953,71 @@ private unsafe void INTERNAL_parseEffectStruct(IntPtr effectData)
 							{
 								memSize *= mem[j].info.elements;
 							}
-							memList.Add(new EffectParameter(
+							string cachedString = String.Empty;
+							if (mem[j].info.parameter_type == MOJOSHADER_symbolType.MOJOSHADER_SYMTYPE_STRING)
+							{
+								MOJOSHADER_effectObject* objectsPtr = (MOJOSHADER_effectObject*)effectPtr->objects;
+								int* index = param.value.values + curOffset;
+								cachedString = Marshal.PtrToStringAnsi(objectsPtr[*index].stringvalue.stringvalue);
+							}
+
+
+							EffectParameter toAdd = new EffectParameter(
 								Marshal.PtrToStringAnsi(mem[j].name),
 								null,
-								(int) mem[j].info.rows,
-								(int) mem[j].info.columns,
-								(int) mem[j].info.elements,
-								XNAClass[(int) mem[j].info.parameter_class],
-								XNAType[(int) mem[j].info.parameter_type],
+								(int)mem[j].info.rows,
+								(int)mem[j].info.columns,
+								(int)mem[j].info.elements,
+								XNAClass[(int)mem[j].info.parameter_class],
+								XNAType[(int)mem[j].info.parameter_type],
 								null, // FIXME: Nested structs! -flibit
 								null,
 								param.value.values + curOffset.ToInt32(),
 								memSize * 4
-							));
+							);
+							toAdd.cachedString = cachedString;
+
+							memList.Add(toAdd);
+							//if(param.value.type.parameter_type == MOJOSHADER_symbolType.MOJOSHADER_SYMTYPE_STRING)
+							//{
+							//	int* index = param.value.values;
+							//	memList[j].cachedString = Marshal.PtrToStringAnsi(objects[*index].stringvalue.stringvalue);
+							//}
+
+							for(int j=0;j<memList.Count;j++)
```

Comment:
Doesn't this block do the same thing as above?

Reply:
Yea forgot to clean up.